### PR TITLE
Fix API problems (Newgrounds!)

### PR DESCRIPTION
**Pull request by David-Orangemoon**
Fixes a problem with logging in!

### CODE REVIEW

File: extensions/obviousAlexC/newgroundsIO.js [hunk 1]
```diff
@@ -1873,48 +2666,44 @@
...
-          xhr.onreadystatechange = function () {
-            if (xhr.readyState == 4) {
-              var o_return;
-              try {
-                o_return = JSON.parse(xhr.responseText);
-              } catch (e) {
-                o_return = { success: false, app_id: core.app_id };
-                o_return.error = { message: String(e), code: 8002 };
-              }
+        var xhr = new XMLHttpRequest();
```
Comment (GarboMuffin) [inline extensions/obviousAlexC/newgroundsIO.js:2669]: still need to check Scratch.canFetch

File: extensions/obviousAlexC/newgroundsIO.js [hunk 2]
```diff
@@ -1933,7 +2722,7 @@
           this.GATEWAY_URI +
           "?request=" +
           encodeURIComponent(JSON.stringify(request));
-        Scratch.openWindow(url);
+        Scratch.openWindow(url, "_blank");
```
Comment (GarboMuffin) [inline extensions/obviousAlexC/newgroundsIO.js:2725]: adding _blank does not do anything there, it is implied

File: extensions/obviousAlexC/newgroundsIO.js [hunk 3]
```diff
@@ -5231,19 +6020,15 @@
...
-          xhr.onreadystatechange = function () {
-            if (xhr.readyState == 4) {
-              if (thisArg) callback.call(thisArg, xhr.responseText);
-              else callback(xhr.responseText);
-            }
-          };
-          xhr.open("GET", this.url, true);
-          xhr.send();
-        });
+        var xhr = new XMLHttpRequest();
```
Comment (GarboMuffin) [inline extensions/obviousAlexC/newgroundsIO.js:6023]: need to check Scratch.canFetch

File: extensions/obviousAlexC/newgroundsIO.js [hunk 4]
```diff
@@ -5883,7 +6668,7 @@
         this.#status = NewgroundsIO.SessionState.WAITING_FOR_USER;
         this.mode = "check";
 
-        Scratch.openWindow(this.passport_url);
+        Scratch.openWindow(this.passport_url, "_blank");
```
Comment (GarboMuffin) [inline extensions/obviousAlexC/newgroundsIO.js:6671]: same again _blank doesn't do anything

**Comment by David-Orangemoon** [comment]
Ok give me a minute I forgot to use the github version and used the version
on my local hard drive by accident lol.

**Comment by GarboMuffin** [comment]
hope you tested this, i did not

**Comment by David-Orangemoon** [comment]
> hope you tested this, i did not

From what I had seen the login was the issue. I did test the problem spot there since that was what was reported to me!

**Pull request merged**